python/cxx/deck_keyword.cpp

```diff
@@ -56,11 +59,70 @@ struct DeckRecordIterator
     }
 };
 
+
+void push_string_as_deck_value(std::vector<DeckValue>& record, const std::string str) {
+
+    if (str.back() == '*') {
+        int it;
+
+        if (str.length() == 1)
+            it = 1;
+        else
+            it = stod( str.substr(0, str.length() - 1) );
```

Comment:
This should be `stoi` - we are not after a double here. Sorry if I have misunderstood your code previously and sent you in the wrong direction with review comments.